codeforamerica/vita-min
# GYR1-468 - Last Contact column updates even if message is undelivered

## [GYR1-468](https://codeforamerica.atlassian.net/browse/GYR1-468)
## What was done?
 * `last_outgoing_communication_at` is now updated on save rather than create, and only when the message is `completed` / `delivered`
 * Added specs for undelivered interactions
 * Updated specs for delivered interactions
## How to test?
* Test on staging (Because emails do not send on heroku)
* Basically we want to show that the `last contact at` is updated only when messages are delivered.
* Go to a client that has opted in to contact by email, and one that has opted in to contact via sms. The screenshot shows one with a valid email address that has opted in to email ![image](https://github.com/codeforamerica/vita-min/assets/17094895/4979abdd-87a5-4f76-9333-341f083dfe84)
* You can see the welcome email 
* Go to the messages tab ![image](https://github.com/codeforamerica/vita-min/assets/17094895/e73b91de-7fd6-4e57-a52c-bccd5c13284c)
* Fill in the form to send an email ![image](https://github.com/codeforamerica/vita-min/assets/17094895/ffcdd266-3dfb-43db-a8b3-75005fd7a31c)
* The last contact at should not update until the message is successfully delivered 
![image](https://github.com/codeforamerica/vita-min/assets/17094895/552567ed-daac-4265-a082-adb96e057940) 
![image](https://github.com/codeforamerica/vita-min/assets/17094895/ea5dded5-a519-46f5-8815-5101971b8e90)


[GYR1-468]: https://codeforamerica.atlassian.net/browse/GYR1-468?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
